ykrt/src/compile/jitc_llvm.rs

```diff
@@ -0,0 +1,24 @@
+//! An LLVM JIT backend. Currently a minimal wrapper around the fact that [IRTrace]s are hardcoded
+//! to be compiled with LLVM.
+
+use crate::{compile::Compiler, trace::IRTrace};
+use libc::c_void;
+use std::{error::Error, sync::Arc};
+use tempfile::NamedTempFile;
+
+pub(crate) struct JITCLLVM;
+
+impl JITCLLVM {
+    pub(crate) fn new() -> Result<Arc<Self>, Box<dyn Error>> {
+        Ok(Arc::new(JITCLLVM))
+    }
+}
+
+impl Compiler for JITCLLVM {
```

Comment:
You've said yourself that we have to be very specific with names of things in this system. We have many compilation steps in our system, so should `Compiler` be `TraceNativeCodeGenerator` or something similarly very specific (but hopefully shorter)?

Reply:
I umed and ahed on this, but on reflect "AOT compiler" and "JIT compiler" seem like reasonable pairings to me. Do I like "JIT" as a term? Not really. But "TraceNative" doesn't conjure up a useful image to me, and just feels long for the sake of it.

Naming is a nightmare...